app/Image/MediaFile.php

```diff
@@ -378,13 +379,11 @@ public static function assertIsSupportedFileExtension(string $extension): void
 	protected static function getSanitizedAcceptedRawFileExtensions(): array
 	{
 		if (self::$cachedAcceptedRawFileExtensions === null) {
-			$tmp = explode('|', strtolower(Configs::get_value('raw_formats', '')));
+			$tmp = explode('|', strtolower(Configs::getValueAsString('raw_formats', '')));
 			// Explode may return `false` on error
 			// Our supported file extensions always take precedence over any
 			// custom configured extension
-			self::$cachedAcceptedRawFileExtensions = is_array($tmp) ?
-				array_diff($tmp, self::SUPPORTED_IMAGE_FILE_EXTENSIONS, self::SUPPORTED_VIDEO_FILE_EXTENSIONS) :
-				[];
+			self::$cachedAcceptedRawFileExtensions = array_diff($tmp, self::SUPPORTED_IMAGE_FILE_EXTENSIONS, self::SUPPORTED_VIDEO_FILE_EXTENSIONS);
```

Comment:
I skipped all of @ildyria's changes to the files inside `Image/` during my review, because most of it will be replaced by my pending PR for using file streams anyway. So I would not spend too much time on these files.